### ISSUE TITLE
false `no-binary` negative when sub-package does not match SRPM name

### ISSUE THREAD

**Issue by ktdreyer** [rpm-software-management/rpmlint#861]
When linting a sub-package, if the sub-package's name does not match the source package's name, rpmlint will never display a "`no-binary`" error.

### Steps to reproduce:
1. Create an SRPM `ansible-core` that generates two sub-packages: `ansible-core` and `ansible-test`.
2. Make sure neither package is `noarch`.
3. Make sure neither package contains arch-specific binaries. (In other words, the packages *should* be noarch, but they are not.)

### Expected results:
`rpmlint` will display `E: no-binary` for the main `ansible-core` sub-package, but not the `ansible-test` sub-package.

### Actual results:
`rpmlint` will display `E: no-binary` for both `ansible-core` and `ansible-test`.

### Additional information:
In `checks/BinariesCheck.py`, we determine if this RPM is a "multi pkg", like so:

```python
multi_pkg = (pkg.name != srcname.group(1))
```

In other words, if my SRPM is `ansible-core`, and the RPM I'm linting is a sub-package named `ansible-test`, then we consider this a "multi pkg" (`multi_pkg = True`).

Later, in `_check_no_binary()`, we do *not* show `E: no-binary` if `multi_pkg` is `True`.

It seems like we could delete the `multi_pkg` logic entirely from `checks/BinariesCheck.py`. Looking all the way back at `multi_pkg` introduction in 668aaccc05692e2582c750a5ac9bfeeb64a0658b, it's not clear to me why this code exists.

Discovered while working on https://bugzilla.redhat.com/2083717 for ansible in RHEL 9.

**Comment by marxin** [comment]
Thank you for the nice analysis. Yes, I can confirm that it does not make much sense and I'm suggesting removal of `multi_pkg`.

**Comment by ktdreyer** [comment]
Thanks for fixing this!

Incredible to see [the commit that introduced this](https://github.com/rpm-software-management/rpmlint/commit/668aaccc05692e2582c750a5ac9bfeeb64a0658b) is a month shy of 20 years old.

**Comment by socketpair** [comment]
@ktdreyer @marxin 

Finally, I have the problem:

main package is arch-specific (i.e. does NOT have `BuildArch: noarch`). But it's subpackages actually do not depend on any arch (Python subpackage without arch-specific part). It's impossible to specify `noarch` for subpackages (at least I did no find how to).

So, rpmlint says: `no-binary`. How to overcome the problem ?


How to
